packages/devtools_app/lib/src/analytics/prompt.dart

```diff
@@ -0,0 +1,151 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/gestures.dart';
+import 'package:flutter/material.dart';
+
+import '../theme.dart';
+import '../utils.dart';
+import 'provider.dart';
+
+/// Conditionally displays a prompt to request permission for collection of
+/// usage analytics.
+class AnalyticsPrompt extends StatefulWidget {
+  const AnalyticsPrompt({
+    @required this.provider,
+    @required this.child,
+  });
+
+  final Widget child;
+  final AnalyticsProvider provider;
+
+  @override
+  State<AnalyticsPrompt> createState() =>
+      _AnalyticsPromptState(provider, child);
+}
+
+class _AnalyticsPromptState extends State<AnalyticsPrompt> {
+  _AnalyticsPromptState(this._provider, this._child);
+
+  final Widget _child;
+  final AnalyticsProvider _provider;
+
+  bool _isVisible = false;
+
+  @override
+  void initState() {
+    super.initState();
+    () async {
```

Comment:
having an unhandled await in initState is always a bit suspect.
I would suggest awaiting for _provider.initialize() outside of a widget. Ideally all this before any UI is rendered perhaps from main.dart
that way you won't get a flash of 1 frame of content without a prompt. Delaying the first frame rendered for the analytics status wouldn't be a big deal. Another reasonable option would be to move this to a controller.

Reply:
I'll follow up with a PR that fixes this issue. Want to get analytics turned on asap.